infra/fastly/terraform/dl.k8s.io/vcl/binaries.vcl

```diff
@@ -0,0 +1,144 @@
+sub vcl_recv {
+  # configure purges to require api authentication:
+  # https://docs.fastly.com/en/guides/authenticating-api-purge-requests
+  #
+  if (req.method == "FASTLYPURGE") {
+      set req.http.Fastly-Purge-Requires-Auth = "1";
+  }
+
+  # Prevent edge from caching stale content served from shield
+  # https://developer.fastly.com/learning/concepts/stale/#shielding-considerations
+  if (fastly.ff.visits_this_service != 0) {
+    set req.max_stale_while_revalidate = 0s;
+  }
+
+
+#FASTLY recv
+
+  # don't bother doing a cache lookup for a request type that isn't cacheable
+  if (req.method !~ "(GET|HEAD|FASTLYPURGE)") {
+    return(pass);
+  }
+
+  return(lookup);
+}
+
+sub vcl_fetch {
+
+  if (beresp.status >= 200 && beresp.status < 300) {
+    set beresp.cacheable = true;
+  }
+
+  /* handle 5XX (or any other unwanted status code) */
+  if (beresp.status >= 500 && beresp.status < 600) {
+    /* deliver stale if the object is available */
+    if (stale.exists) {
+      return(deliver_stale);
+    }
+
+    if (req.restarts < 1 && (req.request == "GET" || req.request == "HEAD")) {
+      restart;
+    }
+
+    set beresp.cacheable = true;
+    set beresp.ttl = 1s;
+    /* else go to vcl_error to deliver a synthetic */
+  error beresp.status;
+  }
+
+  if (beresp.http.Surrogate-Control !~ "(stale-while-revalidate|stale-if-error)") {
+    set beresp.stale_if_error = 31536000s; // 1 year
+    set beresp.stale_while_revalidate = 60s; // 1 minute
+  }
+
+#FASTLY fetch
+  if ((beresp.status == 500 || beresp.status == 503) && req.restarts < 1 && (req.request == "GET" || req.request == "HEAD")) {
+    restart;
+  }
+
+  if (req.restarts > 0) {
+    set beresp.http.Fastly-Restarts = req.restarts;
+  }
+
+  if (beresp.http.Set-Cookie) {
+    set req.http.Fastly-Cachetype = "SETCOOKIE";
+    return(pass);
+  }
+
+  if (beresp.http.Cache-Control ~ "private") {
+    set req.http.Fastly-Cachetype = "PRIVATE";
+    return(pass);
+  }
+
+  # If we've gotten an error after the restarts we'll deliver the response
+  # with a very short cache time.
+  if (http_status_matches(beresp.status, "500,502,503")) {
```

Comment:
I don't think this `if` block is needed either; it'll follow the synthetic error path above.

Reply:
Removed